### PR TITLE
disallow 2 shadows only for ambassador

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
**What this PR does / why we need it**:

Reason to disallow 2 shadows in https://github.com/SeldonIO/seldon-core/commit/76d9a483938881395fc01843c26ed10ce87d421c was specific to ambassador https://github.com/SeldonIO/seldon-core/issues/2383

This issue allows again to deploy more than one shadow if `ambassador.enabled: false` in the helm values.

**Which issue(s) this PR fixes**:

Fixes https://github.com/SeldonIO/seldon-core/issues/2991

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go
Comment (axsaucedo) [inline operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go:42]: Amend to variable to avoid magic word, same for default value (perhaps add on constants)

**Review by axsaucedo — CHANGES_REQUESTED** [review]
Looks good - may be worth adding some documentation to make it clear that more than 2 shadows works in istio but not in ambassador

**Comment by RafalSkolasinski** [comment]
Closing for now as it seems multiple mirrors are not supported with Istio as well: https://github.com/istio/istio/issues/13330

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | b86c94702b4e87ae3ad790b567341a33c49f42d1 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-bddh2) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | b86c94702b4e87ae3ad790b567341a33c49f42d1 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-bddh2) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | b86c94702b4e87ae3ad790b567341a33c49f42d1 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-bddh2) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Pull request closed without merging**
